src/wymeditor/editor/image-handler.js

```diff
@@ -0,0 +1,681 @@
+    var ih = this;
+
+    if (!evt.target.tagName) {
+        // IE8 may fire such an event.
+        // what element was it fired on?
+        return false;
+    }
+
+    if (ih._resizingNow) {
+        // this is up high in this method for performance
+        ih._resizeImage(evt.clientY);
+        return false;
+    }
+
+    if (
+        evt.target.tagName.toLowerCase() === 'img' &&
+        !ih._isResizeHandleAttached()
+    ) {
+        ih._placeResizeHandleOnImg(evt.target);
+        return false;
+    }
+
+    if (!ih._isResizeHandleAttached()) {
+        return false;
+    }
+
+    // from this point on, this event handler is all about
+    // checking whether the resize handle should be detached
+
+    if (
+        !jQuery(evt.target).hasClass(WYMeditor.EDITOR_ONLY_CLASS) &&
+        !ih._isCurrentImg(evt.target)
+    ) {
+        // this must be after the above check for whether resizing now
+        // because, while the resize operation does begin
+        // with the mouse pointing on the resize handle,
+        // the mouse might leave the resize handle during the resize operation.
+        // in that case, we would like the operation to continue
+        ih._detachResizeHandle();
+        return false;
+    }
+
+    if (!ih._isCurrentImgAtMarker()) {
+        ih._detachResizeHandle();
+        return false;
+    }
+
+    // returning false here would disable image dragging
+};
+
+WYMeditor.ImageHandler.prototype._isCurrentImgAtMarker = function () {
+    var ih = this;
+    var $marker = ih._$currentImageMarker;
+    if (!$marker.length) {
+        // the marker was removed by some DOM manipulation
+        return false;
+    }
+    var $img = ih._$currentImg;
+    var $imgPrevToMarker = $marker.prev('img');
+    if (
+        $img.length &&
+        $imgPrevToMarker.length &&
+        $imgPrevToMarker[0] === $img[0]
+    ) {
+        return true;
+    }
+    // this happens when:
+    //
+    // * the image was selected and
+    //   * replaced by pasted content
+    //   * replaced by character insertion from key press
+    //   * removed with backspace/delete
+    // * caret was before/after image and delete/backspace pressed
+    // * the image was dragged and dropped somewhere
+    return false;
+};
+
+WYMeditor.ImageHandler.prototype._isResizeHandle = function (elem) {
+    return jQuery(elem).hasClass(WYMeditor.RESIZE_HANDLE_CLASS);
+};
+
+WYMeditor.ImageHandler.prototype._isCurrentImg = function (img) {
+    var ih = this;
+    return img === ih._$currentImg[0];
+};
+
+WYMeditor.ImageHandler.prototype._resizeImage = function (currentMouseY) {
+    var ih = this;
+    var $img = ih._$currentImg;
+
+    var dimensionsRatio = $img.data('DimensionsRatio');
+
+    if (!dimensionsRatio) {
+        // in order to prevent dimensions ratio corruption
+        var originalHeight = $img.attr('height');
+        var originalWidth = $img.attr('width');
+        dimensionsRatio =  originalWidth / originalHeight;
```

Comment:
Images with original zero height or width could never reach this code because they can't be `mousemove` over. So that's the end of this little edge case, I guess. Not worth the effort.